externals/llvm-external-projects/torch-mlir-dialects/lib/Conversion/TcpToLinalg/Misc.cpp

```diff
@@ -0,0 +1,100 @@
+//===------------------------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Also available under a BSD-style license. See LICENSE.
+//
+//===----------------------------------------------------------------------===//
+
+#include "torch-mlir-dialects/Conversion/TcpToLinalg/TcpToLinalg.h"
+
+#include "../PassDetail.h"
+#include "PopulatePatterns.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Math/IR/Math.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "torch-mlir-dialects/Dialect/Tcp/IR/TcpDialect.h"
+#include "torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.h"
+
+using namespace mlir;
+using namespace mlir::tcp;
+
+namespace {
+
+void getValuesFromIndexArrayAttribute(ArrayAttr attr,
```

Comment:
You can just create a `SmallVector<int64_t>` within this method and return it instead of return by reference? I think most compilers optimize the return as a move.